validations/webhook.test.ts

```diff
@@ -0,0 +1,336 @@
+        const transferStatusResponse =
+          await fiatConnectClient.getTransferStatus({
+            transferId,
+          })
+        expect(transferStatusResponse.isOk).to.be.true
+        await checkObjectAgainstModel(
+          transferStatusResponse.unwrap(),
+          'TransferStatusResponse',
+        )
+
+        // Webhook Validation
+        expect(
+          !!(config.clientApiKey && config.providerId),
+          'clientApiKey & providerId are required parameters for webhook validation',
+        ).to.be.true
+        await expect(
+          validateTransferWebhook(
+            {
+              transferId,
+              address: wallet.address,
+              status: transferStatusResponse.unwrap().status,
+            },
+            2,
+          ),
+        ).to.not.be.rejected
+      })
+    })
+  }
+  if (config.quoteOutMock) {
+    describe('transfer out', () => {
+      it('sends webhooks for transfer out requests', async () => {
+        // Setup
+        const quoteOutParams = {
+          ...MOCK_QUOTE[config.quoteOutMock],
+          address: wallet.address,
+        }
+        const loginResult = await fiatConnectClient.login()
+        expect(loginResult.isOk).to.be.true
+
+        const addKycResult = await fiatConnectClient.addKyc(mockKYCInfo)
+        expect(addKycResult.isOk).to.be.true
+
+        const addAccountResult = await fiatConnectClient.addFiatAccount(
+          mockAccountData,
+        )
+        expect(addAccountResult.isOk).to.be.true
+        const fiatAccountId = addAccountResult.unwrap().fiatAccountId
+
+        const quoteOutResponse = await fiatConnectClient.createQuoteOut(
+          quoteOutParams,
+        )
+        expect(quoteOutResponse.isOk).to.be.true
+        const quoteOutId = quoteOutResponse.unwrap().quote.quoteId
+
+        const transferOutParams = {
+          idempotencyKey: randomUUID(),
+          data: {
+            fiatAccountId: fiatAccountId,
+            quoteId: quoteOutId,
+          },
+        }
+
+        const transferOutResponse = await fiatConnectClient.transferOut(
+          transferOutParams,
+        )
+        expect(transferOutResponse.isOk).to.be.true
+        expect(transferOutResponse.unwrap().transferStatus).to.be.oneOf([
+          TransferStatus.TransferStarted,
+          TransferStatus.TransferReadyForUserToSendCryptoFunds,
+        ])
+        const transferId = transferOutResponse.unwrap().transferId
+        const transferStatusResponse =
+          await fiatConnectClient.getTransferStatus({
+            transferId,
+          })
+        expect(transferStatusResponse.isOk).to.be.true
+        await checkObjectAgainstModel(
+          transferStatusResponse.unwrap(),
+          'TransferStatusResponse',
+        )
+
+        // Webhook Validation
+        expect(
+          !!(config.clientApiKey && config.providerId),
+          'clientApiKey & providerId are required parameters for webhook validation',
+        ).to.be.true
```

Comment:
this check should probly be at the beginning of the test so it fails as fast as possible

Reply:
same in a couple other places